by-language/java-jooq/README.rst

```diff
@@ -35,24 +35,48 @@ In some kind, jOOQ is similar to `LINQ`_, `but better <Insight into Language
 Integrated Querying_>`_.
 
 
+*******
 Details
-=======
+*******
+
+Overview
+========
+
+The code examples will demonstrate a few of the `different use cases for jOOQ`_.
+That is, how to use the `jOOQ DSL API`_ based on code generated with `jOOQ's
+code generator`_ to take your database schema and reverse-engineer it into a
+set of Java classes, as well how to use the `Dynamic SQL API`_ by using `jOOQ
+as a SQL builder without code generation`_.
 
-The code example will demonstrate a few of the `different use cases for jOOQ`_.
-That is, `Dynamic SQL`_, the `jOOQ DSL API`_, and how to use `jOOQ as a SQL
-builder without code generation`_.
+Schema management
+=================
 
+In many cases, the schema is defined in the form of SQL scripts, which can be
+used with a `database schema migration`_ framework like `Flyway`_,
+`Liquibase`_, `Bytebase`_, etc.
+
+The `DDLDatabase - Code generation from SQL files`_ feature can be used to
+effectively reflect the database schema from SQL DDL files, without needing
+a database instance at all. The code provided within the ``src/generated``
+directory has been generated like this.
 
 Caveats
 =======
 
-- `jOOQ's code generator`_ takes your database schema and reverse-engineers it
-  into a set of Java classes. This feature currently does not work with CrateDB
-  yet. The code provided within the ``src/generated`` directory has not been
-  derived by reflecting the database schema from CrateDB.
+- `jOOQ's code generator`_ currently does not work with directly connecting to
+  a real CrateDB database instance and reflecting the schema from there.
+  Because SQL DDL statements are usually maintained in form of multiple
+  incremental migration scripts anyway, this is considered to be not of a too
+  big concern, see above. With corresponding improvements to CrateDB, this
+  can be made work in the future, see `issue #10 - with reflection from the
+  database`_.
 
 - Most of the jOOQ examples use uppercase letters for the database, table, and
-  field names. CrateDB currently only handles lowercase letters.
+  field names. Within this setup, we have only been able to make it work using
+  lowercase letters.
+
+- We have not been able to make multiple SQL DDL statements work within a
+  single SQL bootstrap file at ``src/main/resources/bootstrap.sql``.
```

Comment:
Is this really a caveat? Isn't defining dedicated SQL files for each table even a cleaner way?

Reply:
I am not sure. I think when using a framework managing your data migrations, there is usually one file per schema-change iteration, containing _multiple_ `ALTER TABLE` statements. This caveat is here as a reminder that I discovered issues in this area, and that I would like to revisit the topic on a subsequent iteration.

Reply:
See also https://github.com/crate/cratedb-examples/pull/11#pullrequestreview-1269127462.

Reply:
This statement has been removed. Apparently, it was based on a quirk on my side, and I have not been able to observe a problem in this area again.